service-runner.js

```diff
@@ -178,19 +204,35 @@ ServiceRunner.prototype._runMaster = function() {
     // Set up rolling restarts
     process.on('SIGHUP', this._rollingRestart.bind(this));
 
-    return this._startWorkers(this.config.num_workers);
+    return this._startWorkers(this.config.num_workers)
+    .then(function(workers) {
+        setTimeout(function() {
+            self._checkHeartbeat();
+        }, 1000);
+        return workers;
+    });
 };
 
-ServiceRunner.prototype._stopWorker = function(id) {
-    var worker = cluster.workers[id];
+ServiceRunner.prototype._stopWorker = function(workerId) {
+    var self = this;
+    var worker = cluster.workers[workerId];
+    self.workerHeartbeatTime[worker.process.pid] = {
```

Comment:
Instead of just setting the value, you should check whether it still exists and if it's `killed` value is set to true before. If these conditions are not satisfied, you should bail out as it means the master is already waiting on the worker to die.